### PR TITLE
Node-RED: fixed tests and added types for normalisePropertyExpression

### PR DESCRIPTION
**Pull request by Shaquu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
https://github.com/node-red/node-red/blob/e5d579c1bb1e7c8fbf2d31b6a72fe7c8a54ea87f/packages/node_modules/%40node-red/util/lib/util.js#L217
https://github.com/node-red/node-red/blob/e5d579c1bb1e7c8fbf2d31b6a72fe7c8a54ea87f/packages/node_modules/%40node-red/editor-client/src/js/ui/utils.js#L733
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by Shaquu** [comment]
@alexk111 have a look please

**Review by tbowmo — APPROVED** [review]

**Comment by Shaquu** [comment]
Ready to merge

File: types/node-red-node-test-helper/index.d.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/node-red/node-red-node-test-helper#readme
 // Definitions by: Alex Kaul <https://github.com/alexk111>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// Minimum TypeScript Version: 3.1
+// Minimum TypeScript Version: 4.7
```
Comment (peterblazejewicz) [inline types/node-red-node-test-helper/index.d.ts:5]: why this specific version? (I could not see a TS feature being added that would require this version bump, though I could be missing this)
if anything you could folks remove this min version requirement altogether, these types are quite based on 'standard' set of TS features.

Comment (Shaquu) [reply types/node-red-node-test-helper/index.d.ts:5]: Well, it was purely empirical.
It is related to express types which are failing with lower TypeScript.

Comment (peterblazejewicz) [reply types/node-red-node-test-helper/index.d.ts:5]: thanks, that legit reason completely.

**Review by peterblazejewicz — APPROVED** [review]

**Comment by Shaquu** [comment]
Ready to merge

**Pull request merged**